src/types/Setting.ts

```diff
@@ -84,11 +89,27 @@ export interface OcpiSetting {
   businessDetails: OcpiBusinessDetails;
 }
 
+export interface OicpSetting {
+  cpo: OicpIdentifier;
+  emsp: OicpIdentifier;
+  currency: string;
+  businessDetails: OicpBusinessDetails;
+}
+
 export interface OcpiIdentifier {
   countryCode: string;
   partyID: string;
 }
 
+// Should be renamed. Certificate and Key are bundled with OperatorID / ProviderID at this moment.
+// Because the roles CPO and EMPSP probably need different certificates to call the Hubject Backend
+export interface OicpIdentifier {
+  countryCode: string;
+  partyID: string;
+  clientCertificate?: string;
+  privateKey?: string;
+}
```

Comment:
Does that mean that each CPO tenant will have to store its certificates key pair in the OICP settings in DB?
If yes, the tenant settings will need a revamp to scale: 
+ One singleton object per tenant settings to cache them and avoid querying the DB;
+ Trigger object with tenant settings refresh when changed in DB.

Reply:
It is not clear at this point. Theoretically each Roaming partner (CPOs & eMSPs) needs own certificates. 
I send a request to Hubject if it is possible to use one certificate for our backend, since all requests from different CPOs come from the same backend server.